[ros2 pkg create] Exporting executable target can cause issues with ament_target_dependencies

## Bug report

- Operating System:
  - Ubuntu 20.04
- Installation type:
  - Docker
- Version or commit hash:
  - ros:foxy
- DDS implementation:
  - Fast-RTPS

#### Steps to reproduce issue
* Use CLI to create package:

        ros2 pkg create --build-type ament_cmake --destination-directory src --library-name test_pkg --node-name test_pkg_node test_pkg
* Create another package that depends on the first:

        ros2 pkg create --build-type ament_cmake --destination-directory src --library-name test_pkg2 --dependencies test_pkg --node-name test_pkg2_node test_pkg2

* Build:

        colcon build --packages-up-to test_pkg2

#### Expected behavior
Both packages are built successfully.

#### Actual behavior
The second package fails to build with CMake issues:

```
--- stderr: test_pkg2                         
CMake Error at CMakeLists.txt:48 (add_executable):
  Target "test_pkg2_node" links to target "test_pkg::test_pkg_node" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:23 (add_library):
  Target "test_pkg2" links to target "test_pkg::test_pkg_node" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:23 (add_library):
  Target "test_pkg2" links to target "test_pkg::test_pkg_node" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:48 (add_executable):
  Target "test_pkg2_node" links to target "test_pkg::test_pkg_node" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

This seems due to the node executable target of the first package being associated to the `export_test_pkg` export
```
install(TARGETS test_pkg_node
  EXPORT export_${PROJECT_NAME}
  DESTINATION lib/${PROJECT_NAME})
```
, which if I understand correctly gets picked up by `ament_target_dependencies` in `test_pkg2`. Removing this association, changing the install call to:
```
install(TARGETS test_pkg_node
  DESTINATION lib/${PROJECT_NAME})
```
resolves this. However I am not sure if there is another reason why that export may be needed.

--- a/ros2pkg/ros2pkg/resource/ament_cmake/CMakeLists.txt.em
+++ b/ros2pkg/ros2pkg/resource/ament_cmake/CMakeLists.txt.em
@@ -80,7 +80,6 @@ ament_target_dependencies(
 @[  end if]@
 
 install(TARGETS @(cpp_node_name)
-  EXPORT export_${PROJECT_NAME}
   DESTINATION lib/${PROJECT_NAME})
 @[end if]@
 

